pkg/runtime/runtime.go

```diff
@@ -1059,10 +1083,34 @@ func (a *DaprRuntime) publishMessageGRPC(msg *pubsub.NewMessage) error {
 	}
 
 	if err != nil {
-		err = fmt.Errorf("error from app while processing pub/sub event: %s", err)
+		errStatus, hasErrStatus := status.FromError(err)
+		if hasErrStatus {
+			if errStatus.Code() == codes.Unimplemented {
+				// DROP
+				log.Warn("non-retriable error returned from app while processing pub/sub event %v: %s", cloudEvent.ID, err)
+				return nil
+			}
+		}
+
+		err = fmt.Errorf("error returned from app while processing pub/sub event %v: %s", cloudEvent.ID, err)
 		log.Debug(err)
 	}
 
+	if res.GetStatus() == 0 {
```

Comment:
I would use switch-case and use enum const in the generated appcallback client -
```
switch res.GetStatus() {
case runtimev1pb.TopicEventResponse_SUCCESS:
case runtimev1pb.TopicEventResponse_RETRY:
case runtimev1pb.TopicEventResponse_DROP:
}
```